python/tests/lang/test_server.py

```diff
@@ -1,45 +1,126 @@
-from unittest import TestCase
-
-from pygls import uris
-from pygls.lsp import methods
-from pygls.lsp.types import (
-    ClientCapabilities,
-    Hover,
-    HoverParams,
-    InitializeParams,
-    Position,
-)
-
-from tests.lang.lsp_test_client import LspTestClient
-
-
-class TestLspServer(TestCase):
-    def setUp(self):
-        self.client = LspTestClient()
-        self.client.start()
-        res = self.client.send_request(
-            methods.INITIALIZE,
-            InitializeParams(
-                process_id=12345, root_uri="file://", capabilities=ClientCapabilities()
-            ),
-        )
-
-        self.assertIn("capabilities", res)
-
-    def tearDown(self):
-        self.client.stop()
-
-    def test_handles_hover_request(self):
-        res: Hover = self.client.send_request(
-            methods.HOVER,
-            HoverParams(
-                text_document={"uri": TEST_DOCUMENT_URI},
-                position=Position(line=0, character=0),
-            ),
-        )
-
-        self.assertSequenceEqual(list(res.keys()), ["contents"])
-        self.assertIn("LSP server", res.get("contents"))
-
-
-TEST_DOCUMENT_URI = uris.from_fs_path(__file__)
+from unittest.async_case import IsolatedAsyncioTestCase
+
+from aac.lang.lsp.code_completion_provider import SPACE_TRIGGER
+
+from tests.helpers.lsp.responses.hover_response import HoverResponse
+from tests.helpers.lsp.responses.completion_response import CompletionResponse
+from tests.lang.base_lsp_test_case import BaseLspTestCase
+
+
+class TestLspServer(BaseLspTestCase, IsolatedAsyncioTestCase):
+    async def asyncSetUp(self) -> None:
+        await super().asyncSetUp()
+        self.active_context = self.client.server.language_context
+        await self.create_document("test.aac", TEST_DOCUMENT_CONTENT)
+
+    async def test_adds_definitions_when_opening_file(self):
+        await self.create_document("added.aac", TEST_DOCUMENT_ADDITIONAL_CONTENT)
```

Comment:
I like the pattern of demonstrating that the definitions aren't in the context prior to being piped in via the LSP.
```suggestion
        self.assertIsNone(self.active_context.get_definition_by_name(TEST_ADDITIONAL_SCHEMA_NAME))             
        self.assertIsNone(self.active_context.get_definition_by_name(TEST_ADDITIONAL_MODEL_NAME))
        
        await self.create_document("added.aac", TEST_DOCUMENT_ADDITIONAL_CONTENT)
```